src/main/resources/db/changelog/ddl/0016_final_retention_data_removed_column.xml

```diff
@@ -0,0 +1,93 @@
+
+        <addColumn tableName="anonymous_elector_document">
+            <column name="initial_retention_data_removed" type="boolean" valueBoolean="false" afterColumn="initial_retention_removal_date"
+                    remarks="Indicates if initial retention period data has been removed from an AED">
+                <constraints nullable="false"/>
+            </column>
+            <column name="final_retention_removal_date" type="date" afterColumn="initial_retention_data_removed"
+                    remarks="The date that all remaining data should be removed after the final retention period">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+
+        <!-- remove existing index based on just source type and initial_retention_removal_date -->
+        <dropIndex tableName="anonymous_elector_document" indexName="aed_initial_retention_removal_date_idx"/>
+
+        <!-- include initial_retention_data_removed column in new index -->
+        <createIndex tableName="anonymous_elector_document" indexName="aed_initial_retention_removal_idx">
+            <column name="source_type"/>
+            <column name="initial_retention_data_removed"/>
+            <column name="initial_retention_removal_date"/>
+        </createIndex>
+
+        <!-- create new index for final_retention_removal_date -->
+        <createIndex tableName="anonymous_elector_document" indexName="aed_final_retention_removal_idx">
+            <column name="source_type"/>
+            <column name="final_retention_removal_date"/>
+        </createIndex>
+
+        <rollback>
+            <dropIndex tableName="anonymous_elector_document" indexName="aed_initial_retention_removal_idx"/>
+            <dropIndex tableName="anonymous_elector_document" indexName="aed_final_retention_removal_idx"/>
+
+            <dropColumn
+                tableName="anonymous_elector_document"
+                columnName="initial_retention_data_removed"/>
+            <dropColumn
+                tableName="anonymous_elector_document"
+                columnName="final_retention_removal_date"/>
+
+            <!-- restore the original index based on source type and initial_retention_removal_date -->
+            <createIndex tableName="anonymous_elector_document" indexName="aed_initial_retention_removal_date_idx">
+                <column name="source_type"/>
+                <column name="initial_retention_removal_date"/>
+            </createIndex>
+        </rollback>
+    </changeSet>
+
+    <changeSet author="matt.wills@valtech.com" id="0016_EIP1-4154_alter_aed elector_roll_number width" context="ddl">
+        <modifyDataType
+            columnName="electoral_roll_number"
+            newDataType="varchar(30)"
+            tableName="anonymous_elector_document"/>
+
+        <rollback>
+            <modifyDataType
+                columnName="electoral_roll_number"
+                newDataType="varchar(7)"
```

Comment:
Similarly `NOT NULL` should be applied during rollback.

Reply:
Ah yes. Fixed 👍